lib/src/int64.dart

```diff
@@ -672,18 +750,34 @@ class Int64 implements IntX {
   @override
   String toString() => _toRadixString(10);
 
-  // TODO(rice) - Make this faster by avoiding arithmetic.
+  /// Reusable buffer used by [toHexString].
+  static Uint8List? _buffer;
+
   @override
   String toHexString() {
     if (isZero) return '0';
-    Int64 x = this;
-    String hexStr = '';
-    while (!x.isZero) {
-      int digit = x._l & 0xf;
-      hexStr = '${_hexDigit(digit)}$hexStr';
-      x = x.shiftRightUnsigned(4);
+    const hexDigits = '0123456789ABCDEF';
+
+    var buffer = _buffer ??= Uint8List(16);
+    var bits = _h;
+    var index = 0;
+    void writeBits(int count) {
+      while (count >= 4) {
+        count -= 4;
+        var char = (bits >> count) & 0xF;
+        if (index > 0 || char != 0) {
+          // Avoid leading zeros.
+          buffer[index++] = hexDigits.codeUnitAt(char);
+        }
+      }
     }
-    return hexStr;
+
+    writeBits(20);
+    bits = _m;
+    writeBits(22);
```

Comment:
It has 22 bits available, but only writes 20 of them (because it writes in chunks of 4 bits). It needs to know the number of bits because it takes them from the top, and leaves the two lsb's at the end.